### PR TITLE
nix-tour: refactor, add desktop entry, 0.0.1 -> unstable-2022-01-03

### PR DESCRIPTION
**Pull request by yuuyins**
###### Description of changes
refactor, add desktop entry, 0.0.1 -> unstable-2022-01-03

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by yuuyins** [comment]
@qknight

File: pkgs/applications/misc/nix-tour/default.nix [hunk 1]
```diff
@@ -1,34 +1,46 @@
...
-    cp -R * $out/share
-    chmod 0755 $out/share/ -R
-    echo "#!${runtimeShell}" > $out/bin/nix-tour
-    echo "cd $out/share/" >> $out/bin/nix-tour
-    echo "${electron}/bin/electron $out/share/electron-main.js" >> $out/bin/nix-tour
-    chmod 0755 $out/bin/nix-tour
+    install -d $out/bin $out/share/${pname}
+    cp -R * $out/share/${pname}
+
+    makeWrapper ${electron}/bin/electron $out/bin/${pname} \
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/nix-tour/default.nix:25]:
```suggestion
    install -d $out/bin $out/share/nix-tour
    cp -R * $out/share/nix-tour

    makeWrapper ${electron}/bin/electron $out/bin/nix-tour \
```

File: pkgs/applications/misc/nix-tour/default.nix [hunk 2]
```diff
@@ -1,34 +1,46 @@
...
+      --add-flags $out/share/${pname}/electron-main.js
   '';
 
+  desktopItems = [
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/nix-tour/default.nix:29]: Missing copyDesktopItems in nativeBuildInputs.

File: pkgs/applications/misc/nix-tour/default.nix [hunk 3]
```diff
@@ -1,34 +1,46 @@
...
+    (makeDesktopItem {
+      name = pname;
+      desktopName = "Tour of Nix";
+      genericName = "Tour of Nix";
+      comment = "Interactive programming guide dedicated to the nix programming language";
+      categories = "Development;Education;ComputerScience;Documentation;";
+      exec = "$out/bin/${pname}";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/nix-tour/default.nix:36]:
```suggestion
      exec = "$out/bin/nix-tour";
```

**Comment by yuuyins** [comment]
@SuperSandro2000 done. btw what's the reason for not using pname in `installPhase` and `desktopItems`?

**Comment by SuperSandro2000** [comment]
> @SuperSandro2000 done. btw what's the reason for not using pname in `installPhase` and `desktopItems`?

Its easier to overlook if not every occurrence of pname is substituted.

**Comment by SuperSandro2000** [comment]
@yuuyins can you please rebase?

**Pull request merged**